modules/core/src/main/scala/com.snowplowanalytics.snowplow.lakes/processing/SparkUtils.scala

```diff
@@ -97,17 +105,21 @@ private[processing] object SparkUtils {
     viewName: String,
     writerParallelism: Int
   ): F[DataFrame] =
-    Sync[F].blocking {
+    Sync[F].delay {
       spark
         .table(viewName)
+        .repartitionByRange(writerParallelism, col("event_name"), col("event_id"))
```

Comment:
Why `event_id` is needed exactly ?

Reply:
There is a danger that repartitioning leads to unequal sized partitions.  For example... imagine we are running on a pod with 8cpu, but this pipeline receives 70% page pings, and 30% of everything else.  One of the output partitions will contain 70% of all the incoming events.  And this is not the best way to make utilize the 8 cpu.  And it becomes more difficult to sink all the events within the timespan of one window.

By adding `event_id`, all of the output partitions have exactly equal size.

Reply:
I see. Also, is usage of `repartitionByRange` with `event_name` and `event_id` field correct ? What I understand from [this doc](https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.DataFrame.repartitionByRange.html), partition column needs to be numeric to be able partition by range.